python/dgl/mock_sparse/diag_matrix.py

```diff
@@ -1,4 +1,5 @@
 """dgl diagonal matrix module."""
+from __future__ import annotations
```

Comment:
The annotations is only available in python 3.7. As DGL supports python 3.6, this perhaps may not be suitable. @BarclayII  what do you think?

Reply:
Understood. But due to the version requirement, we probably should drop this.